### PR TITLE
UNG-1933 fix typo and add validation for csv

### PR DESCRIPTION
**Pull request by Hiroki6**
- added some validations for paring csv
- fix some typos
- added updating status of PoC
- added some error handling

### CODE REVIEW

File: src/main/scala/com/ubirch/services/poc/InformationProvider.scala [hunk 1]
```diff
@@ -49,15 +49,16 @@ class InformationProviderImpl @Inject() (conf: Config)(implicit formats: Formats
...
+      goClientRequest(statusAndPW, body)
         .onErrorHandle(ex => throwAndLogError(status, "an error occurred when providing info to go client; ", ex))
     }
   }
 
   @throws[PocCreationError]
-  protected def goClientRequest(statusAndPW: StatusAndPW, body: String): Future[StatusAndPW] = {
-    Future(
+  protected def goClientRequest(statusAndPW: StatusAndPW, body: String): Task[StatusAndPW] =
+    Task.defer(Task.fromFuture {
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/poc/InformationProvider.scala:59]: it looks complicated, but I chose using Task as much as possible.

File: src/main/scala/com/ubirch/services/poc/InformationProvider.scala [hunk 2]
```diff
@@ -102,13 +103,12 @@ class InformationProviderImpl @Inject() (conf: Config)(implicit formats: Formats
         .map {
           _.code match {
             case Ok =>
-              statusAndPW.pocStatus.copy(goClientProvided = true)
+              statusAndPW.pocStatus.copy(certifyApiProvided = true)
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/poc/InformationProvider.scala:106]: here was typo

File: src/main/scala/com/ubirch/services/poc/InformationProvider.scala [hunk 3]
```diff
@@ -49,15 +49,16 @@ class InformationProviderImpl @Inject() (conf: Config)(implicit formats: Formats
...
 
   @throws[PocCreationError]
-  protected def goClientRequest(statusAndPW: StatusAndPW, body: String): Future[StatusAndPW] = {
-    Future(
+  protected def goClientRequest(statusAndPW: StatusAndPW, body: String): Task[StatusAndPW] =
+    Task.defer(Task.fromFuture {
+      // an error could occur before calls the send() method.
+      // In this case, the defer method is needed because the fromFuture method can't catch such an error.
       basicRequest
         .put(uri"$goClientURL")
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/poc/InformationProvider.scala:63]: is it put? not post?

Comment (lalottacontinua) [reply src/main/scala/com/ubirch/services/poc/InformationProvider.scala:63]: it is actually put. a bit surprisingly, true.

File: src/main/scala/com/ubirch/services/poc/PocCreator.scala [hunk 4]
```diff
@@ -79,16 +85,16 @@ class PocCreatorImpl @Inject() (
...
     creationResult
-      .map { status =>
-        pocStatusTable
-          .updatePocStatus(status)
-          .map(_ => Right(status))
-      }.onErrorHandle(handlePocCreationError).flatten
+      .flatMap { status =>
+        val completePoc = poc.copy(status = Completed)
+        // transaction is not necessary here with PocStatusTable as this part would be retry again.
+        pocTable.updatePoc(completePoc).map(_ => Right(status))
```
[the remark this answers is not in the document]
Comment (lalottacontinua) [reply src/main/scala/com/ubirch/services/poc/PocCreator.scala]: That looks good, yes. I must have forgotten to change this. And two more things I just noticed. 
1) we never set the poc.status from Pending to Processing - will fix this. 

And 2) I update the poc in both methods that are concerned with groups and roles, but use each time the original poc as a parameter. So information added in doUserRealmRelatedTasks probably are reset by doDeviceRealmRelatedTasks. Maybe we can think tomorrow about a good solution. 

I would prefer to solve this, by just handing over the pocTriple to each method and return it as a result. But the unwrapping looks not so nice. Maybe Function.tupled() would be a good way to handle it. Or do you have a better idea?

**Review by lalottacontinua — COMMENTED** [review]
We still need to fix one aspect. Maybe you have a good idea.

File: src/main/scala/com/ubirch/services/poc/InformationProvider.scala [hunk 5]
```diff
@@ -84,15 +84,16 @@ class InformationProviderImpl @Inject() (conf: Config)(implicit formats: Formats
...
+      certifyApiRequest(statusAndPW, body)
         .onErrorHandle(ex => throwAndLogError(status, "an error occurred when providing info to certify api; ", ex))
     }
   }
 
   @throws[PocCreationError]
-  protected def certifyApiRequest(statusAndPW: StatusAndPW, body: String): Future[PocStatus] = {
-    Future(
+  protected def certifyApiRequest(statusAndPW: StatusAndPW, body: String): Task[PocStatus] =
+    Task.defer(Task.fromFuture {
```
[the remark this answers is not in the document]
Comment (Hiroki6) [reply src/main/scala/com/ubirch/services/poc/InformationProvider.scala]: I changed it. thanks.

**Review by lalottacontinua — APPROVED** [review]

**Pull request merged**